src/zmalloc.h

```diff
@@ -104,12 +104,12 @@ __attribute__((malloc)) void *ztrymalloc(size_t size);
 __attribute__((malloc)) void *ztrycalloc(size_t size);
 __attribute__((alloc_size(2))) void *ztryrealloc(void *ptr, size_t size);
 void zfree(void *ptr);
-__attribute__((malloc)) __attribute__((alloc_size(1))) void *zmalloc_usable(size_t size, size_t *usable);
-__attribute__((malloc)) __attribute__((alloc_size(1))) void *zcalloc_usable(size_t size, size_t *usable);
-__attribute__((alloc_size(2))) void *zrealloc_usable(void *ptr, size_t size, size_t *usable);
-__attribute__((malloc)) __attribute__((alloc_size(1))) void *ztrymalloc_usable(size_t size, size_t *usable);
-__attribute__((malloc)) __attribute__((alloc_size(1))) void *ztrycalloc_usable(size_t size, size_t *usable);
-__attribute__((alloc_size(2))) void *ztryrealloc_usable(void *ptr, size_t size, size_t *usable);
+void *zmalloc_usable(size_t size, size_t *usable);
+void *zcalloc_usable(size_t size, size_t *usable);
+void *zrealloc_usable(void *ptr, size_t size, size_t *usable);
+void *ztrymalloc_usable(size_t size, size_t *usable);
+void *ztrycalloc_usable(size_t size, size_t *usable);
+void *ztryrealloc_usable(void *ptr, size_t size, size_t *usable);
```

Comment:
This basically prevents the compiler from seeing the allocations altogether.

Reply:
Basically the allocated size (and the allocation itself) never really escapes these interfaces unless the functions get inlined (e.g. with LTO). As a result, fortification may not happen at all for calls to these functions if LTO is not enabled.

So it might make sense to also test with LTO to be sure that everything is covered.

Reply:
We do test with LTO nowadays (enabled for all out builds in our dev branch and next release, but not in the previous one, in which the problem was uncovered), but I don't think we have any CI that uses the latest compiler (the one in which the problem was uncovered).
Maybe we can add a CI job for that.

Regarding the actual change (dropping the attribute from the zmalloc interface), what do you suggest? Keeping it while the fix that uses the dummy realloc internal to it, would make the fix ineffective (the complier will see the requested size). So the only other option is for the code that uses the allocation to directly use the dummy realloc..

I think we can keep the current code, and maybe add CI with bleeding edge compiler.